GVFS/GVFS.Mount/InProcessMount.cs

```diff
@@ -499,7 +499,8 @@ private void HandleUnmountRequest(NamedPipeServer.Connection connection)
         private void MountAndStartWorkingDirectoryCallbacks(CacheServerInfo cache)
         {
             string error;
-            if (!this.context.Enlistment.Authentication.TryRefreshCredentials(this.context.Tracer, out error))
+            if (!this.context.Enlistment.Authentication.TryAnonymousQuery(this.context.Tracer, this.context.Enlistment) &&
+                !this.context.Enlistment.Authentication.TryRefreshCredentials(this.context.Tracer, out error))

```

Comment:
If `TryAnonymousQuery` succeeds but `TryRefreshCredentials` fails, it looks like `Authentication.IsAnonymous` will be false (because the firs thing `TryGetCredentials` is sets it to false).

Am I understanding that right?  (In general I am having trouble understanding how `IsAnonymous` can ever stay true)

Reply:
Ahh right, not sure what i was thinking